actors/builtin/paych/paych_actor.go

```diff
@@ -323,10 +323,14 @@ func (pca Actor) Collect(rt vmr.Runtime, _ *adt.EmptyValue) *adt.EmptyValue {
 	)
 	builtin.RequireSuccess(rt, codeTo, "Failed to send funds to `To`")
 
+	// even though we will delete the actor, it dosen't hurt to set this here for sanity.
 	rt.State().Transaction(&st, func() interface{} {
 		st.ToSend = big.Zero()
 		return nil
 	})
+
+	rt.DeleteActor(builtin.BurntFundsActorAddr)
```

Comment:
The send to `From` above is redundant, and leaves a possibility of actually burning some funds here if we mess something up later. 

How about replacing the Send to From with setting the From as the beneficiary on deletion here? The transaction setting balance to zero is definitely redundant then.

Reply:
@anorth Why is the send to `From` redundant ?

Reply:
I mean it's redundant if we set From as the beneficiary upon deletion.

Reply:
Done.